steward/src/cellars/aave_v2_stablecoin.rs

```diff
@@ -1,77 +1,143 @@
-use ethers::{abi::AbiEncode, prelude::H160};
+use ethers::{
+    abi::AbiEncode,
+    prelude::{H160, U256},
+};
+use std::convert::TryInto;
 use steward_abi::aave_v2_stablecoin::*;
 use steward_proto::steward::aave_v2_stablecoin::Function::{self, *};
 
-pub fn get_encoded_call(function: Function) -> Vec<u8> {
+use crate::{error::Error, utils::sp_call_error};
+
+const LOG_PREFIX: &str = "AaveV2StablcoinCellar";
+
+pub fn get_encoded_call(function: Function) -> Result<Vec<u8>, Error> {
     match function {
-        EnterStrategy(_) => {
-            let call = EnterStrategyCall {};
-            let call = AaveV2StablecoinCellarCalls::EnterStrategy(call);
-            call.encode()
+        AccrueFees(_) => {
+            let call = AccrueFeesCall {};
+            Ok(AaveV2StablecoinCellarCalls::AccrueFees(call).encode())
+        }
+        ClaimAndUnstake(_) => {
+            let call = ClaimAndUnstakeCall {};
+            Ok(AaveV2StablecoinCellarCalls::ClaimAndUnstake(call).encode())
         }
-        ReinvestAmount(params) => {
-            let call = ReinvestWithAmountCall {
-                amount: params.amount.into(),
+        EnterPosition(_) => {
+            let call = EnterPositionCall {};
+            let call = AaveV2StablecoinCellarCalls::EnterPosition(call);
+            Ok(call.encode())
+        }
+        Rebalance(params) => {
+            // We expect the client to pad the route to length 9
```

Comment:
If we expect the client to both pad out the `route` and `swap_params`, is the contract expecting zero values then to represent "don't do a swap"?

Reply:
Zero values are ignored yes. The `route` and `swapParams` are co-dependent, so you could only have one swap param if you only had three entries in the route array (tokenA, pool, tokenB), two if you had five in route (tokenA, pool1, tokenB, pool2, tokenC), etc.